### PR TITLE
Tickets/dm 24684

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: core/modules/replica/Common.h [hunk 1]
```diff
@@ -135,6 +138,35 @@ class SqlColDef {
...
+
+    SqlIndexColumn(SqlIndexColumn const&) = default;
+    SqlIndexColumn& operator=(SqlIndexColumn const&) = default;
+
+    ~SqlIndexColumn() = default;
+
+
+    std::string name;
+    size_t length = 0;
+    bool ascending = true;
```
Comment (n8pease) [inline core/modules/replica/Common.h:166]: make private with accessors?

File: core/modules/replica/Common.h [hunk 2]
```diff
@@ -246,11 +281,26 @@ class SqlRequestParams {
...
+
     SqlRequestParams() = default;
 
     explicit SqlRequestParams(ProtocolRequestSql const& request);
 
     std::string type2str() const;
+
+    static std::string indexSpec2str(IndexSpec indexSpec);
+    static IndexSpec str2indexSpec(std::string const& str);
+    static ProtocolRequestSql::IndexSpec translateIndexSpec(IndexSpec indexSpec);
```
Comment (n8pease) [inline core/modules/replica/Common.h:303]: members and functions and enums oh my! 
I'd rather see the members at the bottom and private, but 🤷

File: core/modules/replica/Common.h (hunk 1 above)
Comment (iagaponenko) [reply core/modules/replica/Common.h:166]: What you're saying is an option. Though, that would be definitively an "overkill" for this tiny class, which in essence, is just a `struct` with a constructor (handy for things like `std:vector<T>::emplace_back()`).

File: core/modules/replica/HttpSqlIndexModule.cc [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        // Compute indexes just once and save it for analyzing this and other result sets.
+        if (tableNameIdx == numeric_limits<size_t>::max()) {
+            for (size_t idx = 0, num = resultSet.fields.size(); idx < num; ++idx) {
+                auto&& fieldName = resultSet.fields[idx].name;
+                if      (fieldName == "Table")         { tableNameIdx  = idx; }
+                else if (fieldName == "Key_name")      { keyNameIdx    = idx; }
+                else if (fieldName == "Seq_in_index")  { sequenceIdx   = idx; }
+                else if (fieldName == "Column_name")   { columnNameIdx = idx; }
+                else if (fieldName == "Index_comment") { commentIdx    = idx; }
+            }
```
Comment (n8pease) [inline core/modules/replica/HttpSqlIndexModule.cc:134]: consider defining the strings I guess. Also, maybe you want an else clause, to report an unexpected fieldName?

File: core/modules/replica/WorkerSqlRequest.cc [hunk 4]
```diff
@@ -350,6 +367,26 @@ string WorkerSqlRequest::_batchQuery(database::mysql::Connection::Ptr const& con
...
+            for (int i = 0; i < _request.index_columns_size(); ++i) {
+                auto const& key = _request.index_columns(i);
+                if (i != 0) keys += ",";
+                keys += conn->sqlId(key.name());
+                if (key.length() != 0) keys += "(" + to_string(key.length()) + ")";
+                keys += key.ascending() ? " ASC" : " DESC";
+            }
+            return "CREATE " + spec + " INDEX " + conn->sqlId(_request.index_name()) + " ON " + databaseTable +
+                   " (" + keys + ")" + " COMMENT " + conn->sqlValue(_request.index_comment());
+        }
```
Comment (n8pease) [inline core/modules/replica/WorkerSqlRequest.cc:383]: add a blank line between cases

**Pull request merged**